File: app/src/main/java/com/qbrains/tampcolapp/Notification/MyFirebaseMessagingService.java
//package com.qbrains.harina.Notification;
//
//import android.annotation.SuppressLint;
//import android.app.NotificationChannel;
//import android.app.NotificationManager;
//import android.app.PendingIntent;
//import android.content.Context;
//import android.content.Intent;
//import android.os.Build;
//import android.widget.RemoteViews;
//
//import androidx.annotation.NonNull;
//import androidx.core.app.NotificationCompat;
//
//import com.google.firebase.messaging.FirebaseMessagingService;
//import com.google.firebase.messaging.RemoteMessage;
//import com.qbrains.harina.R;
//import com.qbrains.harina.ui.component.HarinaApplications;
//
//public class MyFirebaseMessagingService extends FirebaseMessagingService {
//
//
//    @Override
//    public void onMessageReceived(RemoteMessage remoteMessage) {
//        super.onMessageReceived(remoteMessage);
//
//        if (remoteMessage.getNotification() != null) {
//
//            showNotification(
//                    remoteMessage.getNotification().getTitle(),
//                    remoteMessage.getNotification().getBody());
//        }
//    }
//    private RemoteViews getCustomDesign(String title,
//                                        String message) {
//        @SuppressLint("RemoteViewLayout") RemoteViews remoteViews = new RemoteViews(
//                getApplicationContext().getPackageName(),
//                R.layout.notification);
//        remoteViews.setTextViewText(R.id.title, title);
//        remoteViews.setTextViewText(R.id.message, message);
//        remoteViews.setImageViewResource(R.id.icon,
//                R.drawable.call);
//        return remoteViews;
//    }
//
//    public void showNotification(String title,
//                                 String message) {
//        Intent intent
//                = new Intent(this, InfoActivity.class);
//        // Assign channel ID
//        String channel_id = "notification_channel";
//        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
//        // Pass the intent to PendingIntent to start the
//        // next Activity
//        PendingIntent pendingIntent
//                = PendingIntent.getActivity(
//                this, 0, intent,
//                PendingIntent.FLAG_ONE_SHOT);
//
//        NotificationCompat.Builder builder
//                = new NotificationCompat
//                .Builder(getApplicationContext(),
//                channel_id)
//                .setSmallIcon(R.drawable.logo)
//                .setAutoCancel(true)
//                .setVibrate(new long[]{1000, 1000, 1000,
//                        1000, 1000})
//                .setOnlyAlertOnce(true)
//                .setContentIntent(pendingIntent);
//
//        if (Build.VERSION.SDK_INT
//                >= Build.VERSION_CODES.JELLY_BEAN) {
//            builder = builder.setContent(
//                    getCustomDesign(title, message));
//        } // If Android Version is lower than Jelly Beans,
//        // customized layout cannot be used and thus the
//        // layout is set as follows
//        else {
//            builder = builder.setContentTitle(title)
//                    .setContentText(message)
//                    .setSmallIcon(R.drawable.logo);
//        }
//        // Create an object of NotificationManager class to
//        // notify the
//        // user of events that happen in the background.
//        NotificationManager notificationManager
//                = (NotificationManager) getSystemService(
//                Context.NOTIFICATION_SERVICE);
//        // Check if the Android Version is greater than Oreo
//        if (Build.VERSION.SDK_INT
//                >= Build.VERSION_CODES.O) {
//            NotificationChannel notificationChannel
//                    = new NotificationChannel(
//                    channel_id, "web_app",
//                    NotificationManager.IMPORTANCE_HIGH);
//            notificationManager.createNotificationChannel(
//                    notificationChannel);
//        }
//
//        notificationManager.notify(0, builder.build());
//    }
//
//    @Override
//    public void onNewToken(@NonNull String s) {
//        ((HarinaApplications)getApplication()).setFirebaseAppId(s);
//    }
//}

File: app/src/main/java/com/qbrains/tampcolapp/data/firebase/MyFirebaseMessagingService.kt
package com.qbrains.tampcolapp.data.firebase

import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.media.AudioAttributes
import androidx.core.app.NotificationCompat
//import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.qbrains.tampcolapp.R
import com.qbrains.tampcolapp.ui.DashBoardActivity
import com.qbrains.tampcolapp.ui.component.HarinaApplications

class MyFirebaseMessagingService : FirebaseMessagingService() {
    override fun onNewToken(p0: String) {
//        if (PreferenceManager(context = this).getDeviceToken().isNullOrEmpty()) {
//            FirebaseInstanceId.getInstance().instanceId.addOnSuccessListener {
//                it.token
//            }.addOnFailureListener {
//
//            }
//        }
        (application as HarinaApplications).setFirebaseAppId(p0)

    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        println("GET____MESSAGE_____" + remoteMessage.data)
        val data: Map<String, String> = remoteMessage.data
        var title: String = data.getValue("title")
        var message: String = data.getValue("message")
        var intent: Intent? = null


        intent = Intent(applicationContext, DashBoardActivity::class.java)
//            intent.putExtra(NOTIFICATION,KEY_NOTIFICATION)


        val pi: PendingIntent = PendingIntent.getActivity(applicationContext, 101, intent, PendingIntent.FLAG_IMMUTABLE)

        val notificationManager: NotificationManager =
            applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager


        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            var channel: NotificationChannel?
            val audiAt = AudioAttributes.Builder().setUsage(AudioAttributes.USAGE_NOTIFICATION)
                .setContentType(AudioAttributes.CONTENT_TYPE_SPEECH).build()
            channel = NotificationChannel("100", "MyChannel", NotificationManager.IMPORTANCE_HIGH)
            notificationManager.createNotificationChannel(channel)
        }


        var bigTextStyle: NotificationCompat.BigTextStyle? =
            NotificationCompat.BigTextStyle()
        bigTextStyle?.setBigContentTitle(title)
        bigTextStyle?.bigText(message)
        val builder = NotificationCompat.Builder(applicationContext, "100")
            .setContentTitle(title)
            .setContentText(message)
//            .setLargeIcon(R.mipmap.ic_launcher)
            // .setCustomBigContentView(bigView)

            .setAutoCancel(true)
            //.setLargeIcon(ge)
//            .setContentText(data.getValue(NOTIFICATION_MESSAGE))
            .setSmallIcon(R.mipmap.ic_launcher)
//            .setStyle(bigTextStyle)
            //.setContentText(body)
            .setContentIntent(pi)
        builder.priority = NotificationCompat.PRIORITY_HIGH
        notificationManager.notify(101, builder.build())
    }
}